ENG-505 [DRAFT] local development setup

lionakhnazarov opened the pull request:
Add initial setup for local T network development

- Created `genesis.json` for blockchain initialization with predefined accounts and balances.
- Added setup scripts (`setup-geth.sh`, `generate-genesis.sh`, `check-deployments.sh`, `deploy-ecdsa-tbtc.sh`) to facilitate local development and contract deployment.
- Introduced documentation for building and running the T network locally, including quick start guides and local development instructions.
- Added stub contracts for testing and development purposes, ensuring a complete local environment setup.

This commit lays the groundwork for developers to easily set up and interact with the T network in a local environment.

lionakhnazarov committed:
Add initial setup for local T network development

lionakhnazarov committed:
local development setup

lionakhnazarov committed:
stage 3 - status challenge

lionakhnazarov committed:
Add wallet and bridge configuration for Ethereum integration

- Updated  and  to include new addresses for WalletRegistry, Bridge, MaintainerProxy, and WalletProposalValidator.
- Introduced new JSON files for deposit data and funding transaction information to support deposit and redemption processes.
- Added various scripts for managing DKG processes, wallet requests, and operator registration, enhancing local development capabilities.
- Included documentation updates for deployment and operational guidelines.

These changes improve the setup for Ethereum interactions and streamline the development workflow.

lionakhnazarov committed:
local development setup guide

jose-blockchain commented on nohup.out:
```diff
@@ -0,0 +1,13 @@
+==========================================
```
This file seems like local dev output that shouldn't be in the repo.

lionakhnazarov committed:
Enhance RandomBeacon configuration and wallet request scripts

- Added a step in the complete-reset script to fix and authorize the RandomBeacon configuration.
- Updated the request-new-wallet script to check RandomBeacon authorization and wallet creation state, providing detailed error handling and guidance for potential issues.
- Improved logging for better visibility into the DKG state and wallet request process.

These changes streamline the setup and ensure proper configuration for wallet requests and RandomBeacon integration.

lionakhnazarov committed:
reimbursement pool deploy

lionakhnazarov committed:
Update configuration and enhance DKG processes

- Changed Electrum server URL in config.toml for local development.
- Updated various Ethereum contract addresses in configuration files to reflect new deployments.
- Enhanced DKG process scripts to improve error handling and logging.
- Added new methods for fee estimation and transaction handling in the Bitcoin chain integration.
- Improved deposit data structures and scripts for better handling of wallet requests and approvals.

These changes streamline local development and ensure proper integration with updated contract addresses and DKG processes.

lionakhnazarov committed:
Update go.mod and go.sum to replace keep-common with threshold-network version

- Added replacement for github.com/keep-network/keep-common to use github.com/threshold-network/keep-common v1.7.1-tlabs.0 in go.mod.
- Updated go.sum to reflect the new dependency and removed the previous version of keep-common.

lionakhnazarov committed:
Enhance performance metrics and tests

- Updated  to include a new peer for the sepolia network.
- Added timeout handling in  to prevent indefinite hangs.
- Introduced new system metrics: CPU load, RAM utilization, and swap utilization, with corresponding updates to the performance metrics registration.

lionakhnazarov committed:
Enhance coordination window metrics tracking

- Introduced a new  structure to track detailed metrics for individual coordination windows, including timing, success rates, and fault statistics.
- Enhanced the coordination layer to record the start and end of coordination windows, as well as wallet-specific coordination details.
- Added new metrics for coordination windows, including total wallets coordinated, successful, and failed, along with fault tracking.

lionakhnazarov committed:
Add redemption metrics and improve performance tracking

- Introduced new metrics for redemption actions, including total executions, success, and failure counts, as well as duration tracking.
- Updated the performance metrics registration to include these new redemption metrics.
- Refactored existing code to utilize defined constants for metric names, enhancing consistency and readability.
- Improved error handling in redemption proof submissions to accurately record failure metrics.

lionakhnazarov committed:
Enhance coordination window metrics and error handling

- Updated the  and  structures to include JSON tags for improved serialization.
- Introduced a new  structure to capture detailed fault information during coordination.
- Enhanced the  method to include error messages for failed wallet actions.
- Added a new method  to retrieve a summary of coordination window metrics.
- Registered coordination windows as a diagnostic source in the client info for better monitoring.

lionakhnazarov committed:
Implement unique peer ping tracking in libp2p

- Added a mutex and a map to track peers that have already been pinged to avoid duplicate ping tests.
- Updated the connected and disconnected callback functions to manage the pinged peers set, ensuring each unique peer is only pinged once.
- Enhanced disconnection handling to allow re-pinging if a peer reconnects later.

lionakhnazarov committed:
merge

lionakhnazarov committed:
Update configuration and enhance redemption handling in stubs

- Updated BridgeAddress in config.toml and its backup to reflect the new address.
- Modified WalletProposalValidatorStub to include validation functions for redemption proposals.
- Enhanced the BridgeStub with redemption request handling, including event emissions and wallet state checks.
- Improved error logging in redemption tasks and added detailed event filtering for DKG state checks.
- Adjusted coordination parameters for development, including increased block intervals and signing attempts.
